src/Domain/Entity/Promotion.php

```diff
@@ -28,22 +30,29 @@ class Promotion
     /** @var bool */
     private $isSuperPromotion;
 
+    /** @var RelatedLink[]|null */
+    private $relatedLinks;
+
     public function __construct(
         Pid $pid,
         PromotableInterface $promotedEntity,
         string $title,
         Synopses $synopses,
         string $url,
         int $weighting,
-        bool $isSuperPromotion
+        bool $isSuperPromotion,
+        ?array $relatedLinks = null
     ) {
+        $this->assertArrayOfType('related links', $relatedLinks, RelatedLink::class);
```

Comment:
Would https://secure.php.net/manual/en/function.assert.php be appropriate here?

Reply:
Neat, I didn't realise that let you throw exceptions in PHP7. Use of assertions sounds like a good plan for these cases where we can't cover our expectations fully using the type system.

We do this in a several other places (e.g. CollapsedBroadcast, CoreEntity, Version). I'd rather make that change to use asserts all in one go rather than piecemeal. Reckon it's worth creating a tech debt ticket to move everything over?

Reply:
If you like. Can you pop a link to the ticket in a comment to this thread?

Reply:
https://jira.dev.bbc.co.uk/browse/PROGRAMMES-5848